alcs-api/OracleToPg/README.MD

```diff
@@ -0,0 +1,266 @@
+```
+
+- set DATA_PUMP_DIRECTORY in CDB. '/u01/app/oracle/admin/ORCL/dpdump' is a location in the Oracle container where the dump file is.
+
+```sh
+# connect to oracle CDB inside container
+sqlplus sys/Oradoc_db1@localhost:1521/ORCLCDB
+```
+
+- you can check the current connection by executing
+
+```SQL
+show con_name;
+```
+
+```SQL
+-- in sqlplus connected to CDB
+CREATE OR REPLACE DIRECTORY "DATA_PUMP_DIR" as '/u01/app/oracle/admin/ORCL/dpdump';
+```
+
+- switch to the PDB in sqlplus
+
+```SQL
+alter session set container=ORCLPDB1;
+```
+
+- create a table space in PDB
+
+```sql
+CREATE TABLESPACE OATS_TABLES
+   DATAFILE 'OATS_TABLES.dbf'
+   SIZE 1m
+   AUTOEXTEND ON;
+```
+
+- create a PDB import user, where 'IMPOATS' is username, 'Oradoc_db1' is password
+
+```sql
+-- USER SQL
+CREATE USER "IMPOATS" IDENTIFIED BY "Oradoc_db1"
+DEFAULT TABLESPACE "OATS_TABLES"
+TEMPORARY TABLESPACE "TEMP";
+
+-- QUOTAS
+ALTER USER "IMPOATS" QUOTA UNLIMITED ON "SYSTEM";
+ALTER USER "IMPOATS" QUOTA UNLIMITED ON "SYSAUX";
+ALTER USER "IMPOATS" QUOTA UNLIMITED ON "OATS_TABLES";
+ALTER USER "IMPOATS" QUOTA UNLIMITED ON "USERS";
+
+-- ROLES
+GRANT "DBA" TO "IMPOATS" WITH ADMIN OPTION;
+GRANT "DATAPUMP_EXP_FULL_DATABASE" TO "IMPOATS" WITH ADMIN OPTION;
+GRANT "CONNECT" TO "IMPOATS" WITH ADMIN OPTION;
+GRANT "DATAPUMP_IMP_FULL_DATABASE" TO "IMPOATS" WITH ADMIN OPTION;
+```
+
+- start importing data with impdp
+
+```sh
+impdp IMPOATS/Oradoc_db1@localhost:1521/ORCLPDB1.localdomain directory=DATA_PUMP_DIR dumpfile=expdp.OATS-395.20220831.dmp full=y
+```
+
+## Oracle to Postgres migration
+
+Dockerized version of ora2pg tool will be used for migration from Oracle to Postgres. Follow this link for more details: https://github.com/Guy-Incognito/ora2pg
+
+- make sure that the steps above regarding the Oracle data import is completed successfully, and your Oracle instance is running. Use docker ps to check running containers
+
+```sh
+docker ps
+```
+
+- locate 'ora2pg.conf' in the source code. Make sure that the DBI connection string matches your Oracle instance configuration. You are interested in the following parameters: "ORACLE_DSN", "ORACLE_USER", "ORACLE_PWD". NOTE: You want to use OATS user as "ORACLE_USER" for migrating OATS DB.
+- first, prepare the migration of tables schema
+
+  - check ora2pg.conf and make sure that TYPE is set to TABLE
+  - check ora2pg.conf and make sure that FKEY_DEFERRABLE is 1
+  - run ora2pg container:
+
+  ```sh
+   docker run \
+   --name ora2pg \
+   --network <network specified for oracle container above> \
+   -it \
+   -v <path on the host to config folder with ora2pg.conf>:/config \
+   -v <path on the host to data folder where you will see the output of migration>:/data \
+   georgmoser/ora2pg
+
+   # example of command above with populated values
+   docker run \
+   --name ora2pg \
+   --network alpine-net \
+   -it \
+   -v /Users/mekhti/Documents/Bit3/sources/alcs/alcs-api/OracleToPg/ora2pg/config:/config \
```

Comment:
Might want to genericize this or put in a placeholder rather than your username

Reply:
I've added the template/explanation above, but sure I can update it so this one will be more generic. I'll just use 'user' instead